js/ace.js

```diff
@@ -0,0 +1,408 @@
+                '1h': '1h',
+                '2h': '2h',
+                '4h': '4h',
+                '8h': '8h',
+                '12h': '12h',
+                '1d': '1d',
+                '1w': '1w',
+                '1M': '1M',
+            },
+            'urls': {
+                'logo': '',
+                'api': {
+                    'public': 'https://ace.io/polarisex/oapi',
+                    'private': 'https://ace.io/polarisex/open',
+                },
+                'www': 'https://ace.io/',
+                'doc': [
+                    'https://github.com/ace-exchange/ace-offical-api-docs',
+                ],
+                'fees': 'https://helpcenter.ace.io/hc/zh-tw/articles/360018609132-%E8%B2%BB%E7%8E%87%E8%AA%AA%E6%98%8E',
+            },
+            'requiredCredentials': {
+                'apiKey': true,
+                'secret': true,
+                'phone': true,
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'list/tradePrice',
+                        'list/marketPair',
+                        'list/orderBooks/{base}/{quote}',
+                    ],
+                },
+                'private': {
+                    'post': [
+                        'v1/coin/customerAccount',
+                        'v1/kline/getKlineMin',
+                        'v1/order/order',
+                        'v1/order/cancel',
+                        'v1/order/getOrderList',
+                        'v1/order/showOrderStatus',
+                        'v1/order/showOrderHistory',
+                    ],
+                },
+            },
+            'fees': {
+                'trading': {
+                    'percentage': true,
+                    'maker': this.parseNumber ('0.0005'),
+                    'taker': this.parseNumber ('0.001'),
+                },
+            },
+            'options': {
+                'networks': {
+                    'ERC20': 'ERC20',
+                    'ETH': 'ERC20',
+                    'TRX': 'TRX',
+                    'TRC20': 'TRX',
+                },
+            },
+            'precisionMode': TICK_SIZE,
+            'exceptions': {
+                'exact': {
+                },
+                'broad': {
+                },
+            },
+            'commonCurrencies': {
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name ace#fetchMarkets
+         * @description retrieves data on all markets for ace
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[object]} an array of objects representing market data
+         */
+        let response = await this.publicGetListMarketPair ();
+        // temporary workaround to finx json string
+        if (response.indexOf ('\'') >= 0) {
+            // this line didn't work in php
```

Comment:
The line above or below?

Reply:
The line below, I used regex. They'll fix api response.